Rm3 roulette

cp-ugorji opened the pull request:
Write components implementing HTML markup for the required design for the home page.
According to the requirement, I haven't implemented API calls and routing.

I also tried to Implement and use the <ErrorBoundary /> component for catching and displaying errors.

Dmitry-White commented on roulette/package.json:
```diff
@@ -0,0 +1,53 @@
+{
```
As was mentioned in the previous PR: please get rid of the redundant folder nesting.
In your repository "netflix-react" at the root level (i.e. at the level of `netflix-react/`) there should be:
- only 1 pakcage.json
- only 1 pakcage-lock.json
- only 1 .eslintrc, 1 .gitignore, 1 README.md, etc
There's no need to keep a separate copy of these files under several unnecessary folders.

Move the contents of `netflix-react/roulette/*` to `netflix-react/`.
Move the contents of `netflix-react/rm1/src/components/*` to `netflix-react/misc`
Remove rm1, rm2 and roulette folders completely.

Dmitry-White commented on roulette/package.json:
```diff
@@ -0,0 +1,53 @@
+    "eslint": "eslint .",
+    "eslint:fix": "eslint --fix",
+    "prettier": "prettier --check .",
+    "prettier:fix": "prettier --write './**/*.{js,jsx,ts,tsx,css,md,json}'"
```
You're missing "lint" and "lint:fix" commands to check both ESLint & Prettier and check & auto fix both ESLint & Prettier respectively.

Dmitry-White replied:
You're missing "lint" and "lint:fix" commands to check both ESLint & Prettier and check & auto fix both ESLint & Prettier respectively.

Dmitry-White commented on roulette/package.json:
```diff
@@ -0,0 +1,53 @@
+    "build": "react-scripts build",
+    "test": "react-scripts test",
+    "eject": "react-scripts eject",
+    "eslint": "eslint .",
```
Since you forgot to lint your code this time before pushing your changes it would be more practical to add some automation instead.
Therefore, consider adding Husky pre-push hook to the project.
https://github.com/typicode/husky

Dmitry-White replied:
Since you forgot to lint your code this time before pushing your changes it would be more practical to add some automation instead.
Therefore, consider adding Husky pre-push hook to the project.
https://github.com/typicode/husky

Dmitry-White commented on roulette/public/manifest.json:
```diff
@@ -0,0 +1,25 @@
+{
+  "short_name": "React App",
+  "name": "Create React App Sample",
```
Starting from the Task 2 the rest of the tasks are done without CRA.
That was the entire purpose of learning React + Webpack configuration from Task 2.
Therefore, consider using your setup from Task 2.

You essentially have 2 options:
- copy all the related React code from `netflix-react/roulette/src` to your `netflix-react/rm2`
- copy all the related project configuration from `netflix-react/rm2/*` to your `netflix-react/roulette/`

Dmitry-White replied:
Starting from the Task 2 the rest of the tasks are done without CRA.
That was the entire purpose of learning React + Webpack configuration from Task 2.
Therefore, consider using your setup from Task 2.

You essentially have 2 options:
- copy all the related React code from `netflix-react/roulette/src` to your `netflix-react/rm2`
- copy all the related project configuration from `netflix-react/rm2/*` to your `netflix-react/roulette/`

Dmitry-White commented on roulette/src/components/modals/MovieForm.js:
```diff
@@ -0,0 +1,10 @@
+
+// import { DialogContent, DialogOverlay } from '@reach/dialog';
```
Consider not leaving commented code in the repo.

Dmitry-White replied:
Consider not leaving commented code in the repo.

Dmitry-White commented on roulette/src/components/movieItem/Movie.js:
```diff
@@ -0,0 +1,42 @@
+
+      <div className="movie-container">
+        <div>
+          <img src={movie.image_url} height="461" width="337" />
```
Consider moving all the sizing to CSS.

Dmitry-White replied:
Consider moving all the sizing to CSS.

Dmitry-White commented on roulette/src/App.js:
```diff
@@ -0,0 +1,32 @@
+import MovieSearch from './components/movieSearch/MovieSearch';
+import movieData from './data/movies';
+import ErrorBoundary from './components/movieError/ErrorBoundary';
+import MovieSort from './components/movieSort/MovieSort';
```
Consider grouping imports of the same level together (./oneThing and ../anotherThing are different levels) and separating each level with an empty line. Additionally, order imports top-down from most-global to most-local - it's usually:
- public library imports
- private library imports
- imports of different levels within the project (level ~= the number of "../")
- local imports for thing that are in the same folder as current file)
E.g. https://github.com/import-js/eslint-plugin-import/blob/main/docs/rules/order.md

Additionally, consider adding this plugin to your project to automate that.

Dmitry-White replied:
Consider grouping imports of the same level together (./oneThing and ../anotherThing are different levels) and separating each level with an empty line. Additionally, order imports top-down from most-global to most-local - it's usually:
- public library imports
- private library imports
- imports of different levels within the project (level ~= the number of "../")
- local imports for thing that are in the same folder as current file)
E.g. https://github.com/import-js/eslint-plugin-import/blob/main/docs/rules/order.md

Additionally, consider adding this plugin to your project to automate that.

Dmitry-White commented on roulette/src/components/movieError/ErrorBoundary.js:
```diff
@@ -0,0 +1,33 @@
+  render() {
+    const { error } = this.state;
+    if (error) {
+      return <MovieError error={error} />;
```
Since you already have a working Error Boundary solution, consider also addressing the error that the project has at the moment.
<img width="854" alt="Screenshot 2022-05-03 at 14 18 04" src="https://user-images.githubusercontent.com/25229823/166460189-bdf19cc3-4397-455c-aabd-5ad92be9fb91.png">

This is after running:
`cd netflix-react/roulette`
`npm i`
`npm start`

My Node.js version is 16.14.1

Dmitry-White commented on roulette/src/components/movieError/ErrorBoundary.js:
```diff
@@ -0,0 +1,33 @@
+  render() {
+    const { error } = this.state;
+    if (error) {
+      return <MovieError error={error} />;
```
Since you already have a working Error Boundary solution, consider also addressing the error that the project has at the moment.
<img width="854" alt="Screenshot 2022-05-03 at 14 18 04" src="https://user-images.githubusercontent.com/25229823/166460189-bdf19cc3-4397-455c-aabd-5ad92be9fb91.png">

This is after running:
`cd netflix-react/roulette`
`npm i`
`npm start`

My Node.js version is 16.14.1

cp-ugorji replied:
guess this has something to do with the reach ui library I used, but I was able to remove the error by 
setting strictExportPresence: false in the webpack config in the node_module which I cant push to the repo.

Dmitry-White commented on roulette/src/components/movieList/MovieList.js:
```diff
@@ -0,0 +1,31 @@
+const MovieList = ({ movies }) => {
+  return (
+    <>
+      {movies.length ? (
```
Couple of points here:

1. Consider always creating a separate named function for method callbacks/handlers.
E.g. `list.map(mapHandler);`
2. Consider avoiding putting logic in JSX directly, use function calls or variable instead.
The difference between function call and variable would be the difference in placement relative to the component body itself: function calls for things from the outside and variable for things inside the body.
E.g. `const renderGenres = props.genres.map(renderGenre);`
`<> ... {renderGenres}... </>`

Dmitry-White replied:
Couple of points here:

1. Consider always creating a separate named function for method callbacks/handlers.
E.g. `list.map(mapHandler);`
2. Consider avoiding putting logic in JSX directly, use function calls or variable instead.
The difference between function call and variable would be the difference in placement relative to the component body itself: function calls for things from the outside and variable for things inside the body.
E.g. `const renderGenres = props.genres.map(renderGenre);`
`<> ... {renderGenres}... </>`

Dmitry-White reviewed (changes requested):
The previous PR comments are not used here.
Consider double checking those previous comments as well as addressing new one for this PR.

Dmitry-White commented:
@cp-ugorji 
Could you mark the already addressed comments so I know when I need to re-review everything?

cp-ugorji commented:
Yes sure, I am trying to address the error, I found it's related to the
reach ui I am using and I had to change some configuration on webpack
config in node_module to remove the error on the ui. Although I still see
the error on the command line but now out of the ui however, that won't
help much because the node_module isn't pushed to the repo.

That is my main challenge now because I can view it.

Should I use a different library?

On Thu, May 5, 2022 at 1:24 PM Dmitry White ***@***.***>
wrote:

> @cp-ugorji <https://github.com/cp-ugorji>
> Could you mark the already addressed comments so I know when I need to
> re-review everything?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/cp-ugorji/netflix-react/pull/3#issuecomment-1118485704>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AHZQES46QAQ4OKC2LHJTL7LVIO4X5ANCNFSM5UYEJNJQ>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>


-- 
Best Regards

Dmitry-White commented:
> Yes sure, I am trying to address the error, I found it's related to the reach ui I am using and I had to change some configuration on webpack config in node_module to remove the error on the ui. Although I still see the error on the command line but now out of the ui however, that won't help much because the node_module isn't pushed to the repo. That is my main challenge now because I can view it. Should I use a different library?
> […](#)
> On Thu, May 5, 2022 at 1:24 PM Dmitry White ***@***.***> wrote: @cp-ugorji <https://github.com/cp-ugorji> Could you mark the already addressed comments so I know when I need to re-review everything? — Reply to this email directly, view it on GitHub <[#3 (comment)](https://github.com/cp-ugorji/netflix-react/pull/3#issuecomment-1118485704)>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/AHZQES46QAQ4OKC2LHJTL7LVIO4X5ANCNFSM5UYEJNJQ> . You are receiving this because you were mentioned.Message ID: ***@***.***>
> -- Best Regards

Yeah, a different library would be an easier solution then.
Take a look at the popularity metrics to make the choice in a bit safer way :)

Dmitry-White reviewed (changes requested):
Not all the comments are resolved.
Please review all the comments and address them with related code changes.
Things to look into:
1. Project's still using CRA -> change according to Task 2 PR (webpack, eslintrc, prettierrc, no ESLint & Prettier configs in package.json)
2. Redundant Comments -> remove
3. Missing ESLint & Prettier configs -> running `npm run lint` with incorrect import order or missing semicolon or incorrect indentation should result in error

cp-ugorji commented:
> Not all the comments are resolved. Please review all the comments and address them with related code changes. Things to look into:
> 
> 1. Project's still using CRA -> change according to Task 2 PR (webpack, eslintrc, prettierrc, no ESLint & Prettier configs in package.json)
> 2. Redundant Comments -> remove
> 3. Missing ESLint & Prettier configs -> running `npm run lint` with incorrect import order or missing semicolon or incorrect indentation should result in error

I think I have broken the code, it no longer loading and I can't seem to figure it out yet.

Dmitry-White reviewed (approved):
Approving with an agreement to address the rest of the linting issues in the next PR.
